### DISCUSSION TITLE
[question] need a guide for checking updater locally

### DISCUSSION THREAD

**Discussion by devhimanshuhc** [tauri-apps/tauri#9097] [Q&A]
So i have been working with tauri app, and i am trying to check `Updater` in tauri. I am currently under development and want to implement  the app updation. How can i do so if i am working locally and haven't distributed yet?

**Comment by mikew** [comment]
Easiest would probably be to stand up a static server that hosts the `.json` file and the builds, then point your app to that.

**Reply by CalvinWilkinson** [reply]
I have been trying to do this as well.

I tried to do this by creating a simple deno server locally. I then set up the updater endpoint in the _**tauri.conf.json**_ file to point to the local server. 

When I run the application, nothing happens.

**Reply by devhimanshuhc** [reply]
try to use GitHub gist to send the json file. might work

**Reply by mikew** [reply]
Aye, Gist works very well (just make sure you point to the "latest" version of the file, not one in a specific commit). I used that for a couple of weeks while ironing out my CI workflow.

**Reply by mikew** [reply]
@CalvinWilkinson If you hook into the JS updater (instead of the Tauri supplied one), you can get a better look into what errors are happening.

**Reply by CalvinWilkinson** [reply]
I have been looking into that but the docs are quite minimal. They show the API, and I understand it, but I just don't know where that code should go. 

Do you have any info how how to set it up more than what it shows in the docs?

**Reply by devhimanshuhc** [reply]
ngl it took me a while to understand the actual process of the distribution and updation

**Reply by CalvinWilkinson** [reply]
@mikew I have actually figured this out.  What I did was create a proxy that the updater points to and have the proxy return the correct build asset for the builder to update.

It was a little bit of work but it was not that difficult.  I have the requirement to keep the assets and repository private for my client.  So far this has been working out pretty well.
